lib/storage/file.js

```diff
@@ -872,11 +920,17 @@ File.prototype.startResumableUpload_ = function(stream, metadata) {
       }),
       qs: {
         name: that.name,
-        uploadType: 'resumable'
+        uploadType: 'resumable',
       },
       headers: headers,
       json: metadata
-    }, function(err, res, body) {
+    };
+
+    if (that.explicitGeneration) {
+      reqOpts.qs.ifGenerationMatch = that.explicitGeneration;
```

Comment:
This means: Only update the metadata of the remote file if the current generation of that file matches the generation of this File instance.